cmd/firewall.go

```diff
@@ -173,3 +194,163 @@ func (c *firewallCmd) createRequestFromCLI() (*models.V1FirewallCreateRequest, e
+
+// for test
+// args = [firewall id, identity, control plane address, auth key]
+func (c *config) firewallSSH(args []string) (err error) {
+	if len(args) < 1 {
+		return fmt.Errorf("Machine ID is expected as an argument")
+	}
+	firewallID := args[0]
+
+	// Get firewall's project
+	machineGetResp, err := c.client.Machine().FindMachine(machine.NewFindMachineParams().WithID(firewallID), nil)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve firewall details: %w", err)
+	}
+	projectID := machineGetResp.Payload.Allocation.Project
+
+	authKeyResp, err := c.client.VPN().GetVPNAuthKey(vpn.NewGetVPNAuthKeyParams().WithBody(&models.V1VPNRequest{
+		Pid: projectID,
+	}), nil)
+	if err != nil {
+		return fmt.Errorf("failed to get VPN auth key: %w", err)
+	}
+
+	cli, err := dockerclient.NewClientWithOpts(dockerclient.FromEnv)
+	if err != nil {
+		return fmt.Errorf("failed to initialize Docker client: %w", err)
+	}
+
+	// Deploy tailscaled
+	ctx := context.Background()
+	containerConfig := &container.Config{
+		Image: "tailscale/tailscale",
+		Cmd:   []string{"tailscaled"},
+	}
+	hostConfig := &container.HostConfig{
+		Binds: []string{
+			"/var/lib:/var/lib",
+			"/dev/net/tun:/dev/net/tun",
+		},
+		NetworkMode: container.NetworkMode("host"),
+		Privileged:  true,
+	}
+	containerName := "tailscaled"
+	resp, err := cli.ContainerCreate(ctx, containerConfig, hostConfig, nil, nil, containerName)
+	if err != nil {
+		return c.describePrinter.Print(err)
+	}
+
+	tailscaledContainerID := resp.ID
+	defer func() {
+		if e := cli.ContainerRemove(ctx, tailscaledContainerID, types.ContainerRemoveOptions{}); e != nil {
+			if err != nil {
+				e = fmt.Errorf("%s: %w", e, err)
+			}
+			err = c.describePrinter.Print(e)
+		}
+	}()
+
+	if err = cli.ContainerStart(ctx, tailscaledContainerID, types.ContainerStartOptions{}); err != nil {
+		return c.describePrinter.Print(err)
+	}
+	defer func() {
+		if e := cli.ContainerStop(ctx, tailscaledContainerID, nil); e != nil {
+			if err != nil {
+				e = fmt.Errorf("%s: %w", e, err)
+			}
+			err = c.describePrinter.Print(e)
+		}
+	}()
+
+	// Exec tailscale up
+	execConfig := types.ExecConfig{
+		Cmd: []string{"tailscale", "up", "--auth-key=" + *authKeyResp.Payload.AuthKey, "--login-server=" + *authKeyResp.Payload.Address},
+	}
+	execResp, err := cli.ContainerExecCreate(ctx, containerName, execConfig)
+	if err != nil {
+		return c.describePrinter.Print(
+			fmt.Errorf("failed to create tailscaled exec: %w", err),
+		)
+	}
+	if err := cli.ContainerExecStart(ctx, execResp.ID, types.ExecStartCheck{}); err != nil {
+		return c.describePrinter.Print(
+			fmt.Errorf("failed to start tailscaled exec: %w", err),
+		)
+	}
+
+	// Connect to the firewall via SSH
+	firewallVPNAddr, err := c.getFirewallVPNAddr(ctx, cli, containerName, *machineGetResp.Payload.Allocation.Hostname)
+	if err != nil {
+		return fmt.Errorf("failed to get Firewall VPN address")
+	}
+
+	err = SSHClient("metal", viper.GetString("identity"), firewallVPNAddr, 22)
```

Comment:
Identity path needs to be expanded with home directory. Should be somewhere in the project already.